implementations/elixir/ockam/ockam/lib/ockam/api/response.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Ockam.API.Response do
+  @moduledoc """
+  Okcam request-response API response
+  """
+
+  alias Ockam.API.Request
+
+  defstruct [:id, :request_id, :status, body: "", from_route: [], to_route: []]
+
+  @max_id 65_534
+
+  @schema {:map, [:id, :request_id, :status, :has_body]}
+
+  def encode(response) when is_map(response) do
+    body =
+      case response.body do
+        binary when is_binary(binary) -> binary
+        ## TODO fail instead?
```

Comment:
agree, +1 to specify   response.body  as `nil | binary()`   and fail otherwise.    Makes it simpler to reason about it  (what can/can't be put as body).   Plus when decoding back with `decode/1` ,  the  body is not decoded. 
 `x ==  decode(encode(x))` is great to maintain

Reply:
How do we treat the "" in that case? Do we say that `has_body=true` and it's empty?
@twittner some feedback from the Rust side is also useful from that point. Do we want to have `have_body=true` and empty bodies?

Reply:
For now let's set the body to `nil` if `has_body=false` and to `""` if `has_body=true` and there is no body.